=======================================================================
Tests for Class: BaseGeometry
=======================================================================

Class:
======

'BaseGeometry' should create an instance of it's type

::
	>>> import sys
	>>> sys.path.append('..')
	>>> BaseGeometry = __import__("7-base_geometry").BaseGeometry
	>>> shape = BaseGeometry()
	>>> isinstance(shape, BaseGeometry)
	True


Method: area(self)
==============

'BaseGeometry' should have the method 'area()'

::
	>>> callable(getattr(shape, "area"))
	True

The method should not be implemented

::
	>>> shape.area()
	Traceback (most recent call last):
	Exception: area() is not implemented


Method: integer_validator(self, name, value)
=============================================
'BaseGeometry' should have the method 'integer_validator'

::
	>>> callable(getattr(shape, "integer_validator"))
	True

'integer_validator' has to take two arguments

::
	>>> shape.integer_validator()
	Traceback (most recent call last):
	TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'
	
	>>> shape.integer_validator('number')
	Traceback (most recent call last):
	TypeError: integer_validator() missing 1 required positional argument: 'value'

'value' must be an integer greater than 0

::
	>>> shape.integer_validator('number', 1)

	>>> shape.integer_validator('number', 0)
	Traceback (most recent call last):
	ValueError: number must be greater than 0

	>>> shape.integer_validator('number', -4)
	Traceback (most recent call last):
	ValueError: number must be greater than 0

	>>> shape.integer_validator('number', 'number')
	Traceback (most recent call last):
	TypeError: number must be an integer

	>>> shape.integer_validator('number', (2,))
	Traceback (most recent call last):
	TypeError: number must be an integer

	>>> shape.integer_validator('number', [2, 4])
	Traceback (most recent call last):
	TypeError: number must be an integer

	>>> shape.integer_validator('number', True)
	Traceback (most recent call last):
	TypeError: number must be an integer

	>>> shape.integer_validator('number', None)
	Traceback (most recent call last):
	TypeError: number must be an integer

	>>> shape.integer_validator('number', {2, 1})
	Traceback (most recent call last):
	TypeError: number must be an integer
